src/main/java/BooleanFlagRefactoring.java

```diff
@@ -0,0 +1,153 @@
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.List;
+import org.eclipse.jdt.core.dom.AST;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.ConditionalExpression;
+import org.eclipse.jdt.core.dom.Expression;
+import org.eclipse.jdt.core.dom.IfStatement;
+import org.eclipse.jdt.core.dom.InfixExpression;
+import org.eclipse.jdt.core.dom.NullLiteral;
+import org.eclipse.jdt.core.dom.ParenthesizedExpression;
+import org.eclipse.jdt.core.dom.PrimitiveType;
+import org.eclipse.jdt.core.dom.SimpleName;
+import org.eclipse.jdt.core.dom.Type;
+import org.eclipse.jdt.core.dom.VariableDeclarationFragment;
+import org.eclipse.jdt.core.dom.VariableDeclarationStatement;
+import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
+
+/**
+ * This class represents a refactoring in which boolean flags are replaced with
+ * explicit null checks
+ */
+public class BooleanFlagRefactoring extends Refactoring {
+	public static final String NAME = "BooleanFlagRefactoring";
+
+	/**
+	 * List of variable names identified as boolean flags, along with their
+	 * corresponding initializer expression
+	 */
+	private final Dictionary<String, Expression> booleanFlags;
+
+	/** Default constructor (for RefactoringEngine integration) */
+	public BooleanFlagRefactoring() {
+		super();
+		this.booleanFlags = new Hashtable<>();
+	}
+
+	@Override
+	public boolean isApplicable(ASTNode node) {
+
+		if (node instanceof VariableDeclarationStatement stmt) {
+			return isApplicable(stmt);
+		}
+		if (node instanceof IfStatement ifStmt) {
+			return isApplicable(ifStmt);
+		}
+		return false;
+
+	}
+
+	/**
+	 * Checks to see if a VariableDeclarationStatement defines a boolean flag that
+	 * represents another variable's nullness
+	 */
+	public boolean isApplicable(VariableDeclarationStatement stmt) {
+		Type stmtType = stmt.getType();
+		boolean isBooleanDeclaration = (stmtType instanceof PrimitiveType pType
+				&& pType.getPrimitiveTypeCode() == PrimitiveType.BOOLEAN);
+
+		// Must be boolean variable
+		if (!isBooleanDeclaration) {
+			return false;
+		}
+
+		boolean flagFound = false;
+
+		// Search through all declared variables in declaration node for a booleanflag
+		for (int i = 0; i < stmt.fragments().size(); ++i) {
+			VariableDeclarationFragment frag = (VariableDeclarationFragment) stmt.fragments().get(i);
```

Comment:
This loop looks like it could be an enhanced for instead - I don't think you need `i` except for iteration.